### PR TITLE
Cooking Codex

### PR DESCRIPTION
**Pull request by DreamySkrell**
image

image

### CODE REVIEW

**Comment by DreamySkrell** [comment]
!review

**Comment by DreamySkrell** [comment]
image
1. remove this
2. add program to civilian presets only
3. test if everything works nicely

**Comment by DreamySkrell** [comment]
should be ok now

**Review by NonQueueingMatt — APPROVED** [review]

**Comment by DreamySkrell** [comment]
changed reagents to be like `10u sugar` instead of `10x sugar`

File: code/__DEFINES/subsystem-priority.dm [hunk 1]
```diff
@@ -26,10 +26,11 @@
 #define SS_INIT_LIGHTING   	0	// Generation of lighting overlays and pre-bake. May cause openturf updates, should initialize before SSzcopy.
 #define SS_INIT_ZCOPY      -1	// Z-mimic flush. Should run after SSoverlay & SSicon_smooth so it copies the smoothed sprites.
 #define SS_INIT_XENOARCH   -2   // Xenoarch is this far below because it can infinite loop if placed in SS_INIT_MISC as it was before, due to some subsystems spawning stuff there.
-#define SS_INIT_HOLOMAP    -3   // Minimap subsystem. Should be initialized after all maps, away sites, ships, planets, etc.
-#define SS_INIT_LOBBY      -4	// Lobby timer starts here. The lobby timer won't actually start going down until the MC starts ticking, so you probably want this last
-#define SS_INIT_PING 	   -5	// Pinger for the clients
-#define SS_INIT_CHAT       -6	// To ensure chat remains smooth during init.
+#define SS_INIT_CODEX      -3   // Codex subsystem. Should be initialized after chemistry and cooking recipes.
```
Comment (FluffyGhoster) [inline code/__DEFINES/subsystem-priority.dm:29]: They go into `code\__DEFINES\subsystems.dm` as `INIT_ORDER_*` (as we update them), this is legacy init defines that will disappear as things gets updated

File: code/controllers/subsystems/initialization/codex.dm [hunk 2]
```diff
@@ -0,0 +1,121 @@
+
+SUBSYSTEM_DEF(codex)
+	name = "Codex"
+	flags = SS_NO_FIRE
+	init_order = SS_INIT_CODEX
```
Comment (FluffyGhoster) [inline code/controllers/subsystems/initialization/codex.dm:5]: As above

File: code/controllers/subsystems/initialization/codex.dm [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+	var/list/chemistry_codex_data = list()
+	var/list/chemistry_codex_ignored_reaction_path = list(/datum/chemical_reaction/slime)
+	var/list/chemistry_codex_ignored_result_path = list(/singleton/reagent/drink, /singleton/reagent/alcohol)
+
+/datum/controller/subsystem/codex/Initialize()
+	generate_cooking_codex()
+	generate_chemistry_codex()
+
+	LOG_DEBUG("SScodex: [cooking_codex_data.len] cooking recipes; [chemistry_codex_data.len] chemistry recipes.")
```
Comment (FluffyGhoster) [inline code/controllers/subsystems/initialization/codex.dm:18]: This needs its own subsystem to log, if it needs to log anything; See a file in `code\__HELPERS\logging\subsystems\` for reference

**Review by Geevies — APPROVED** [review]

**Pull request merged**